Sources/StreamChatUI/Composer/ComposerVC.swift

```diff
@@ -796,6 +813,47 @@ open class ComposerVC: _ViewController,
             log.assertionFailure(error.localizedDescription)
         }
     }
+    
+    open func checkChannelCountdown() {
+        if content.countdownDuration > 0 {
+            var duration = content.countdownDuration
+            let timer = Timer.scheduledTimer(
+                withTimeInterval: 1,
+                repeats: true
+            ) { [weak self] timer in
+                guard let self = self else { return }
+                if duration == 0 {
+                    timer.invalidate()
+                        
+                    self.composerView.countdownButton.isHidden = true
+                        
+                    if self.content.state == .edit {
+                        self.composerView.confirmButton.isHidden = false
+                    } else {
+                        self.composerView.sendButton.isHidden = false
+                    }
+                        
+                    self.composerView.inputMessageView.textView.placeholderLabel.text = L10n.Composer.Placeholder.message
+                    self.composerView.isUserInteractionEnabled = true
+                } else {
+                    self.composerView.sendButton.isHidden = true
+                    self.composerView.countdownButton.isHidden = false
+                    self.composerView.confirmButton.isHidden = true
+                        
+                    self.composerView.inputMessageView.textView.placeholderLabel.text = L10n.Composer.Placeholder.slowMode
+                    self.composerView.isUserInteractionEnabled = false
+                        
+                    self.composerView.countdownButtonWidthConstraint.constant = duration >= 10 ? 40 : 32
+                        
+                    self.composerView.countdownButton.setTitle("\(duration)", for: .disabled)
+                        
+                    duration -= 1
+                }
+            }
+            
+            RunLoop.current.add(timer, forMode: .common)
+        }
+    }
```

Comment:
I think we can break this function into different responsibilities, so that is easier to read and manage:
- Move the logic of the timer to the `CountdownButton`, where we could just call `start(with: countdown)` and it would manage the timer and update the text in the button.
-  Introduce a new `ComposerState.slowMode`, and then the managing of the buttons to be hidden/shown etc will go into the `updateContent()` where we check the `content.state`.

Reply:
Done